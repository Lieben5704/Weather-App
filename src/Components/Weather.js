import React from "react"
import Table from 'react-bootstrap/Table';

let Weather = (props) => {
    return (
        <div>
            <div>
            <Table striped bordered hover>
        <thead>
        <tr>
          <th>Area</th>
          <th>Discription</th>
          <th>Temperature</th>
          <th>Humidity</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>{props.data.name}</td>
          <td>{props.data.weather[0].description}</td>
          <td>{props.data.main.temp}</td>
          <td>{props.data.main.humidity}%</td>
        </tr>
      </tbody>
            </Table>
            </div>
        </div>
    );
} 
export default Weather