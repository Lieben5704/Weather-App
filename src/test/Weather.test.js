import React from 'react';
import renderer from 'react-test-renderer';
import Weather from '../Components/Weather';

describe('<Weather />', () => {
  const mockData = {
    name: 'Cape Town',
    weather: [{ description: 'Sunny' }],
    main: { temp: 20, humidity: 60 },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Weather data={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});