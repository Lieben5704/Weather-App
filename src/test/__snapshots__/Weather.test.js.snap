// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather /> renders correctly 1`] = `
<div>
  <div>
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Area
          </th>
          <th>
            Discription
          </th>
          <th>
            Temperature
          </th>
          <th>
            Humidity
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Cape Town
          </td>
          <td>
            Sunny
          </td>
          <td>
            20
          </td>
          <td>
            60
            %
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
